common/io/src/unmarshal.rs

```diff
@@ -14,6 +14,9 @@
 
 pub trait Unmarshal<T> {
     fn unmarshal(scratch: &[u8]) -> T;
+    fn try_unmarshal(scratch: &[u8]) -> Option<T> {
+        Some(Self::unmarshal(scratch))
+    }
```

Comment:
Is there any reason for using Option instead of Result?

Here using `Result<T, E>` is more convenient to express what's going wrong.

The error type could be just the ErrorCode or a type parameter with default value ErrorCode.

Reply:
> However, the reverse is not true: not all valid u32s are valid chars. from_u32() will return None if the input is not a valid value for a char
As I see rust use Option type for covert u32 to char, so I use Option type here.